app/javascript/packages/document-capture/components/file-input.jsx

```diff
@@ -168,6 +168,9 @@ function FileInput(props, ref) {
     const file = /** @type {FileList} */ (event.target.files)[0];
     if (file) {
       if (isValidForAccepts(file.type, accept)) {
+        if (inputRef.current) { 
```

Comment:
This seems to work fine because it makes the title of the input the file name, But I had thought that input files natively tie in the file name so when we read the input it will show the proper file name, That doesnt seem to be working on the React version of the app. I couldn't find an step we may have missed when calling the onChange method in the app either. Idk if theres a more elegant solution im missing.

Reply:
Hm, looking at this component again, I wonder if the problem is related to the logic we have above:

https://github.com/18F/identity-idp/blob/2535c2c07b660bf153593beea4210724432788d4/app/javascript/packages/document-capture/components/file-input.jsx#L131-L145

As mentioned in the code comment, we keep the value of the actual DOM element unset at all times, which might explain why the browser announces it as "no file selected".

I also worry this wouldn't handle mobile, where we "set" a value from what's captured by the camera, but it doesn't actually go through the input element, so this logic is never run, and the input would still technically be empty.

Using `title` is interesting as well, since I could see it as a way to set the label of the field (maybe `aria-label` would be better), though I'm confused if we want to be controlling the label, or controlling the value, since in the [name, role, value structure](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html), I'd consider this more like the value. That being said, I have hard time finding information about [what role a `<input type="file">` actually is](https://w3c.github.io/html-aam/#el-input-file), so maybe label is what we'd want 🤷  We ought to test it in other screen readers though.

If we change this to return as part of the rendered result, rather than assigning it to the input when handling the change event, we can handle both desktop and mobile upload:

```jsx
<input ref={inputRef} type="file" aria-label={getLabelFromValue(value)} {/* ... */} />
```

`getLabelFromValue` would need to handle (a) no value, (b) value of type `File` ([`value.name`](https://developer.mozilla.org/en-US/docs/Web/API/File/name)), and (c) a string value. For a string value, maybe we just default it to something like "captured image", since it represents an image captured with a camera.

Dunno if we might also want to explicitly set a [role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) to something to try to make sure it's announced consistently across devices.

Reply:
Yea. I think you caught exactly where that issue might be coming from. I think for now label makes the most sense. as for the role. I couldnt find a role that made sense other than just input. so ill add that for now